$NetBSD: patch-Makefile,v 1.3 2024/11/27 07:04:55 khorben Exp $

Let pkgsrc determine the search paths, instead of forcing
the prefix.

Set the correct ID for the shared libraries on macOS.

--- Makefile.orig	2022-12-09 11:37:27.000000000 +0000
+++ Makefile
@@ -286,7 +286,7 @@ INCLUDE ?= -I/usr/include
 else ifeq ($(YQ2_OSTYPE),FreeBSD)
 INCLUDE ?= -I/usr/local/include
 else ifeq ($(YQ2_OSTYPE),NetBSD)
-INCLUDE ?= -I/usr/X11R7/include -I/usr/pkg/include
+INCLUDE ?= -I/usr/X11R7/include
 else ifeq ($(YQ2_OSTYPE),OpenBSD)
 INCLUDE ?= -I/usr/local/include
 else ifeq ($(YQ2_OSTYPE),Windows)
@@ -301,7 +301,7 @@ LDFLAGS ?= -L/usr/lib
 else ifeq ($(YQ2_OSTYPE),FreeBSD)
 LDFLAGS ?= -L/usr/local/lib
 else ifeq ($(YQ2_OSTYPE),NetBSD)
-LDFLAGS ?= -L/usr/X11R7/lib -Wl,-R/usr/X11R7/lib -L/usr/pkg/lib -Wl,-R/usr/pkg/lib
+LDFLAGS ?= -L/usr/X11R7/lib
 else ifeq ($(YQ2_OSTYPE),OpenBSD)
 LDFLAGS ?= -L/usr/local/lib
 else ifeq ($(YQ2_OSTYPE),Windows)
@@ -516,7 +516,7 @@ endif
 
 ifeq ($(WITH_RPATH),yes)
 ifeq ($(YQ2_OSTYPE), Darwin)
-release/quake2 : LDFLAGS += -Wl,-rpath,'@executable_path/lib'
+release/quake2 : LDFLAGS += -Wl,-rpath,'$(PREFIX)/lib'
 else
 release/quake2 : LDFLAGS += -Wl,-z,origin,-rpath='$$ORIGIN/lib'
 endif
@@ -579,6 +579,7 @@ ref_gl1:
 
 
 release/ref_gl1.dylib : LDFLAGS += -shared -framework OpenGL
+release/ref_gl1.dylib : LDFLAGS += -Wl,-install_name,$(PREFIX)/libexec/yquake2/ref_gl1.dylib
 
 else # not Windows or Darwin
 
@@ -619,6 +620,7 @@ ref_gl3:
 
 release/ref_gl3.dylib : GLAD_INCLUDE = -Isrc/client/refresh/gl3/glad/include
 release/ref_gl3.dylib : LDFLAGS += -shared
+release/ref_gl3.dylib : LDFLAGS += -Wl,-install_name,$(PREFIX)/libexec/yquake2/ref_gl3.dylib
 
 else # not Windows or Darwin
 
@@ -668,6 +670,7 @@ release/ref_gles3.dylib : GLAD_INCLUDE =
 release/ref_gles3.dylib : CFLAGS += -DYQ2_GL3_GLES3 -DYQ2_GL3_GLES
 
 release/ref_gles3.dylib : LDFLAGS += -shared
+release/ref_gles3.dylib : LDFLAGS += -Wl,-install_name,$(PREFIX)/libexec/yquake2/ref_gles3.dylib
 
 else # not Windows or Darwin
 
@@ -711,6 +714,7 @@ ref_soft:
 	$(MAKE) release/ref_soft.dylib
 
 release/ref_soft.dylib : LDFLAGS += -shared
+release/ref_soft.dylib : LDFLAGS += -Wl,-install_name,$(PREFIX)/libexec/yquake2/ref_soft.dylib
 
 else # not Windows or Darwin
 
